proposed/config-meta.md

```diff
@@ -0,0 +1,134 @@
+# Config Meta Document
+
+## 1. Introduction
+This document describes the process and discussions that led to the Config PSR. It's goal is to explain the reasons 
+behind each decision.
+
+The word `Container` in this document is to be interpreted as the `ContainerInterface` of the PSR Container proposal.
+
+## 2. Why bother?
+The configuration of instances via factories is not uniform, uses different ways to retrieve and check the needed 
+options and it's not clear which configuration options needs the factory to create the instance.
+
+## 3. Scope
+
+### 3.1 Goals
+The goal set by the Config PSR is to standardize how factories uses a configuration to create instances, support for 
+auto discovery of needed configuration, to reduce boilerplate code and to make it more readable and easier to understand. 
+It can also be used to build the name of the *Container* entry.
+
+### 3.2 Non-goals
+* It's not a goal to define a whole project/library configuration structure.
+* It's not a goal to validate the configuration option values.
+
+## 4. History
+Before submitting the Config PSR to the PHP-FIG, the interfaces was first proposed in a project named 
+[interop-config](https://github.com/sandrokeil/interop-config).
+
+The goal of the project was to provide a test-bed for implementing the interfaces and implementation and to pave the way 
+for the Config PSR.
+
+## 5. Interfaces
+There are six interfaces and some interfaces extends from another interface to ensure the uniform configuration structure. 
+If you confused about the naming of the interfaces or methods, note that I have used the Domain Driven Design approach.
+
+### 5.1 RequiresConfig
+This is the main interface and describes the default configuration structure. Since we have Composer more and more 
+libraries can be combined to create individual projects. So it is only logical to start the configuration structure 
+with a `vendorName`. A vendor has multiple packages so the next structure depth is `packageName`.
```

Comment:
It's a big hassle to start it with a vendor name when most of the packages are provided by framework and the vendor is the same. It feels like unnecessary complication in this case.